src/main/java/de/nplay/moderationbot/embeds/EmbedHelpers.java

```diff
@@ -27,27 +29,35 @@ public static EmbedDTO getEmbedWithTarget(String embedName, EmbedCache embedCach
                 .injectValue("color", color);
     }
 
-    public static MessageEmbed getModlogEmbedHeader(EmbedCache embedCache, Member member) {
+    public static MessageEmbed getModlogEmbedHeader(EmbedCache embedCache, ModlogContext context) {
         var spielersucheAusschlussRolle = ConfigService.get(BotConfig.SPIELERSUCHE_AUSSCHLUSS_ROLLE);
 
-        var roles = member.getRoles().stream()
-                .filter(it -> it.getId().equals(spielersucheAusschlussRolle.orElse("-1")))
-                .map(it -> "<@&%s>".formatted(it.getId()))
-                .reduce((a, b) -> a + " " + b)
-                .orElse("?DEL?");
-
         var embed = embedCache.getEmbed("modlogHeader")
-                .injectValue("username", member.getUser().getEffectiveName())
-                .injectValue("userId", member.getIdLong())
-                .injectValue("avatarUrl", member.getUser().getEffectiveAvatarUrl())
-                .injectValue("roles", roles)
-                .injectValue("createdAt", member.getTimeCreated().getLong(ChronoField.INSTANT_SECONDS))
-                .injectValue("joinedAt", member.getTimeJoined().getLong(ChronoField.INSTANT_SECONDS))
-                .injectValue("color", EmbedColors.DEFAULT)
-                .toEmbedBuilder();
+                .injectValue("username", context.user().getEffectiveName())
+                .injectValue("userId", context.user().getIdLong())
+                .injectValue("avatarUrl", context.user().getEffectiveAvatarUrl())
+                .injectValue("createdAt", context.user().getTimeCreated().getLong(ChronoField.INSTANT_SECONDS))
+                .injectValue("color", EmbedColors.DEFAULT);
 
-        embed.getFields().removeIf(it -> Objects.requireNonNullElse(it.getValue(), "").contains("?DEL?"));
-        return embed.build();
+        String roles;
+        if (context.member() == null) {
+            embed.injectValue("roles", "?DEL?")
+                    .injectValue("joinedAt", "?DEL?");
+        } else {
+            roles = context.member().getRoles().stream()
+                    .filter(it -> it.getId().equals(spielersucheAusschlussRolle.orElse("-1")))
+                    .map(it -> "<@&%s>".formatted(it.getId()))
+                    .reduce((a, b) -> a + " " + b)
+                    .orElse("?DEL?");
+
+            embed.injectValue("roles", roles)
+                    .injectValue("joinedAt", context.member().getTimeJoined().getLong(ChronoField.INSTANT_SECONDS));
+
+        }
+        var builder = embed.toEmbedBuilder();
+
+        builder.getFields().removeIf(it -> Objects.requireNonNullElse(it.getValue(), "").contains("?DEL?"));
+        return builder.build();
```

Comment:
Is this the `roles` string really needed outside the if-statement?

```suggestion
        if (context.member() == null) {
            embed.injectValue("roles", "?DEL?")
                    .injectValue("joinedAt", "?DEL?");
        } else {
            var roles = context.member().getRoles().stream()
                    .filter(it -> it.getId().equals(spielersucheAusschlussRolle.orElse("-1")))
                    .map(it -> "<@&%s>".formatted(it.getId()))
                    .reduce((a, b) -> a + " " + b)
                    .orElse("?DEL?");

            embed.injectValue("roles", roles)
                    .injectValue("joinedAt", context.member().getTimeJoined().getLong(ChronoField.INSTANT_SECONDS));

        }
        var builder = embed.toEmbedBuilder();

        builder.getFields().removeIf(it -> Objects.requireNonNullElse(it.getValue(), "").contains("?DEL?"));
        return builder.build();
```